tests/testutil/router.go

```diff
@@ -19,6 +19,26 @@ type TestRoute struct {
 	T      *testing.T
 }
 
+// create an arbitrary router to a service
+// ie "rio router add
+func CreateRoute(t *testing.T, name, svc, action string) TestRoute {
```

Comment:
Whats difference between `CreateRoute` and `Add` ?

Reply:
Add requires a router to exist. This one is creating the router as well.

Reply:
Main reason for that is so we can have Spec handle the cleanup instead of using `defer` inline. If we are going to go this route, Create should at least just create a `TestRoute` and then call Add on it. The different signatures are also somewhat confusing, we should keep the order/names same as actual CLI call.